spec/concurrent/promise_spec.rb

```diff
@@ -368,6 +368,27 @@ def get_ivar_from_args(opts)
 
         expect(composite).to be_rejected
       end
+
+      it 'preserves ordering of the executed promises' do
+        promise1 = Promise.execute do
+          # resolves after the second promise
+          sleep 0.2
```

Comment:
@hobodave is right. @ianks could you update the spec? I would like to include it since this is indeed the expected behavior.